Add Slots Documentation (#4527)
aws-amplify/docs: src/pages/console/uibuilder/override.mdx

</tbody>
</table>

> ### Nesting Collections

Reviewer: In retrospect, I think we should move this below the other section given that only in the next section, we explain "overrideItems"

> Slots within collections become extremely useful when used to render additional nested collections.  This can be used to generate a collection of posts, each with its own collection of comments.  

Reviewer: ```suggestion
Slots within collections allow you to render additional nested collections. For example, use nested collections to generate a collection of posts, each with its own collection of comments.  
```

In this example, we have two collections: `AmpligramCollection` and `CommentViewCollection`.  These are bound to the `Post` and `Comment` data models, which have a [one-to-many relationship](/console/data/relationships/).  

> By nesting the `CommentViewCollection` within the `comment` slot in the `InstagramCollection`, and using the `overrideItems` prop, we can render each Post with its related Comments.

Reviewer: ```suggestion
By nesting the `CommentViewCollection` within the `comment` slot in the `AmpligramCollection`, and using the `overrideItems` prop, we can render each Post with its related Comments.
```

Author: Applying recommended language adjustment

```jsx

> 

Reviewer: remove empty line

import { 
  AmpligramCollection, 
  CommentViewCollection 
} from './ui-components'

function App() {
  return (
    <div>
       <h1>Collection with Slots</h1>
      <AmpligramCollection overrideItems={({ item, index }) => ({
        comments: <CommentViewCollection items={item.Comments} />
      })} />
    </div>
  );
}

export default App;

> 

Reviewer: remove empty line
